src/mp4box/tenc.rs

```diff
@@ -0,0 +1,398 @@
+
+        // reserved
+        reader.read_u8()?;
+
+        let temp = reader.read_u8()?;
+        let (default_crypt_byte_block, default_skip_byte_block) = if version != 0 {
+            let default_crypt_byte_block = (temp & 0xF0) >> 4;
+            let default_skip_byte_block = temp & 0x0F;
+
+            (
+                Some(default_crypt_byte_block),
+                Some(default_skip_byte_block),
+            )
+        } else {
+            (None, None)
+        };
+
+        // 0x00: not protected
+        // 0x01: protected
+        // 0x02 – 0xFF: reserved
+        let default_is_protected = reader.read_u8()? == 1;
+
+        let default_per_sample_iv_size = reader.read_u8()?;
+
+        let mut default_kid = [0; 16];
+        reader.read_exact(&mut default_kid)?;
+
+        let (default_constant_iv_size, default_constant_iv) =
+            if default_is_protected && default_per_sample_iv_size == 0 {
+                let default_constant_iv_size = reader.read_u8()?;
+                let mut default_constant_iv = [0; 16];
+
+                #[allow(clippy::needless_range_loop)]
+                for i in 0..default_constant_iv_size as usize {
+                    default_constant_iv[i] = reader.read_u8()?;
+                }
+
+                (Some(default_constant_iv_size), Some(default_constant_iv))
+            } else {
+                (None, None)
+            };
+
+        skip_bytes_to(reader, start + size)?;
+
+        Ok(TencBox {
+            default_crypt_byte_block,
+            default_skip_byte_block,
+            default_is_protected,
+            default_per_sample_iv_size,
+            default_kid,
+            default_constant_iv_size,
+            default_constant_iv,
+        })
+    }
+}
+
+impl<W: Write> WriteBox<&mut W> for TencBox {
+    fn write_box(&self, writer: &mut W) -> Result<u64> {
+        let size = self.box_size();
+
+        BoxHeader::new(self.box_type(), size).write(writer)?;
+
+        let version =
+            if self.default_skip_byte_block.is_some() && self.default_crypt_byte_block.is_some() {
+                1
+            } else {
+                0
+            };
+
+        write_box_header_ext(writer, version, 0)?;
+
+        // reserved
+        writer.write_u8(0)?;
+
+        let temp = match (self.default_skip_byte_block, self.default_crypt_byte_block) {
+            (Some(skip), Some(crypt)) => (crypt << 4) | (skip),
+            _ => 0,
+        };
+
+        writer.write_u8(temp)?;
+
+        writer.write_u8(if self.default_is_protected { 1 } else { 0 })?;
+
+        writer.write_u8(self.default_per_sample_iv_size)?;
+
+        writer.write_all(&self.default_kid)?;
+
+        if self.default_is_protected && self.default_per_sample_iv_size == 0 {
+            match (&self.default_constant_iv_size, &self.default_constant_iv) {
+                (Some(size), Some(iv)) => {
+                    writer.write_u8(*size)?;
+                    for i in 0..*size {
```

Comment:
```suggestion
writer.write_all(&iv.as_slice()[0..*size as usize])?;
```
Perhaps instead of looping?

Reply:
Japp